### PR TITLE
[Instrumentation.Http] Change Activity Source Name for HttpWebRequest…

### PR DESCRIPTION
**Pull request by Kielek**
Alternative PR to #3509

## Changes

[Instrumentation.Http] Change Activity Source Name for `HttpWebRequest`s from `OpenTelemetry.HttpWebRequest` to `OpenTelemetry.Instrumentation.HttpWebRequest`.

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* ~~[ ] Design discussion issue #~~
* ~~[ ] Changes in public API reviewed~~

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs [hunk 1]
```diff
@@ -36,7 +36,7 @@ namespace OpenTelemetry.Instrumentation.Http.Implementation
     /// </remarks>
     internal static class HttpWebRequestActivitySource
     {
-        public const string ActivitySourceName = "OpenTelemetry.HttpWebRequest";
+        public const string ActivitySourceName = "OpenTelemetry.Instrumentation.HttpWebRequest";
```
Comment (alanwest) [inline src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs:39]: Just tossing one more idea out. When an instrumentation library defines multiple activity sources like, we could adopt the pattern of `AssemblyName`[.{`SomeAdditionalIdentifier`}]. The `SomeAddionalIdentifier` is optional and would only be used when there are multiple activity sources.

So in this case the two ActivitySources in this library would be:

* `OpenTelemetry.Instrumentation.Http.HttpClient` and
* `OpenTelemetry.Instrumentation.Http.HttpWebRequest`

Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs:39]: @open-telemetry/dotnet-approvers, what do you think? I will be glad to prepare changes, but I would like to have agreement on that. :)

Comment (utpilla) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs:39]: I think this is a good idea.

Comment (cijothomas) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs:39]: agree

Comment (cijothomas) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs:39]: with .NET 7, we wont control the ActivitySource name, as httpclient already creates activity with its ActivitySource.

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs (hunk 1 above)
Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs]: Fixed, for currently supported set of .NET version.

I think that change will be useful as long OTel support .NET 6 and .NET Framework.

**Review by utpilla — APPROVED** [review]

**Review by vishweshbankwar — APPROVED** [review]

**Comment by utpilla** [comment]
Please resolve the merge conflicts.

**Review by cijothomas — APPROVED** [review]

**Pull request merged**